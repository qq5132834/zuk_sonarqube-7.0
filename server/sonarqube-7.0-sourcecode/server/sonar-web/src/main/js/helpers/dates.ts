/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import {
  differenceInDays as _differenceInDays,
  differenceInSeconds as _differenceInSeconds,
  differenceInYears as _differenceInYears,
  isSameDay as _isSameDay,
  parse,
  startOfDay as _startOfDay
} from 'date-fns';

function pad(number: number) {
  if (number < 10) {
    return '0' + number;
  }
  return number;
}

type ParsableDate = string | number | Date;

export function parseDate(rawDate: ParsableDate): Date {
  return parse(rawDate);
}

export function toShortNotSoISOString(rawDate: ParsableDate): string {
  const date = parseDate(rawDate);
  return `${date.getFullYear()}-${pad(date.getMonth() + 1)}-${pad(date.getDate())}`;
}

export function toNotSoISOString(rawDate: ParsableDate): string {
  const date = parseDate(rawDate);
  return date.toISOString().replace(/\..+Z$/, '+0000');
}

export function startOfDay(date: Date): Date {
  return _startOfDay(date);
}

export function isValidDate(date: Date): boolean {
  return !isNaN(date.getTime());
}

export function isSameDay(dateLeft: Date, dateRight: Date): boolean {
  return _isSameDay(dateLeft, dateRight);
}

export function differenceInYears(dateLeft: Date, dateRight: Date): number {
  return _differenceInYears(dateLeft, dateRight);
}

export function differenceInDays(dateLeft: Date, dateRight: Date): number {
  return _differenceInDays(dateLeft, dateRight);
}

export function differenceInSeconds(dateLeft: Date, dateRight: Date): number {
  return _differenceInSeconds(dateLeft, dateRight);
}
