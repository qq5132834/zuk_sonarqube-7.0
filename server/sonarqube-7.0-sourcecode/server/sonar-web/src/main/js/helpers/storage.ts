/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
const PROJECTS_DEFAULT_FILTER = 'sonarqube.projects.default';
const PROJECTS_FAVORITE = 'favorite';
const PROJECTS_ALL = 'all';

const PROJECTS_VIEW = 'sonarqube.projects.view';
const PROJECTS_VISUALIZATION = 'sonarqube.projects.visualization';
const PROJECTS_SORT = 'sonarqube.projects.sort';

const PROJECT_ACTIVITY_GRAPH = 'sonarqube.project_activity.graph';
const PROJECT_ACTIVITY_GRAPH_CUSTOM = 'sonarqube.project_activity.graph.custom';

function save(key: string, value?: string): void {
  try {
    if (value) {
      window.localStorage.setItem(key, value);
    } else {
      window.localStorage.removeItem(key);
    }
  } catch (e) {
    // usually that means the storage is full
    // just do nothing in this case
  }
}

export function saveFavorite(): void {
  save(PROJECTS_DEFAULT_FILTER, PROJECTS_FAVORITE);
}

export function isFavoriteSet(): boolean {
  const setting = window.localStorage.getItem(PROJECTS_DEFAULT_FILTER);
  return setting === PROJECTS_FAVORITE;
}

export function saveAll(): void {
  save(PROJECTS_DEFAULT_FILTER, PROJECTS_ALL);
}

export function isAllSet(): boolean {
  const setting = window.localStorage.getItem(PROJECTS_DEFAULT_FILTER);
  return setting === PROJECTS_ALL;
}

export function saveView(view?: string): void {
  save(PROJECTS_VIEW, view);
}

export function getView(): string | null {
  return window.localStorage.getItem(PROJECTS_VIEW);
}

export function saveVisualization(visualization?: string): void {
  save(PROJECTS_VISUALIZATION, visualization);
}

export function getVisualization(): string | null {
  return window.localStorage.getItem(PROJECTS_VISUALIZATION);
}

export function saveSort(sort?: string): void {
  save(PROJECTS_SORT, sort);
}

export function getSort(): string | null {
  return window.localStorage.getItem(PROJECTS_SORT);
}

export function saveCustomGraph(metrics?: string[]): void {
  save(PROJECT_ACTIVITY_GRAPH_CUSTOM, metrics ? metrics.join(',') : '');
}

export function getCustomGraph(): string[] {
  const customGraphs = window.localStorage.getItem(PROJECT_ACTIVITY_GRAPH_CUSTOM);
  return customGraphs ? customGraphs.split(',') : [];
}

export function saveGraph(graph?: string): void {
  save(PROJECT_ACTIVITY_GRAPH, graph);
}

export function getGraph(): string {
  return window.localStorage.getItem(PROJECT_ACTIVITY_GRAPH) || 'issues';
}
