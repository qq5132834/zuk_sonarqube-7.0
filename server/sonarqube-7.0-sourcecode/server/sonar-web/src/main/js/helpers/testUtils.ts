/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow, ShallowRendererProps, ShallowWrapper, ReactWrapper } from 'enzyme';
import { IntlProvider } from 'react-intl';

export const mockEvent = {
  target: { blur() {} },
  currentTarget: { blur() {} },
  preventDefault() {},
  stopPropagation() {}
};

export function click(element: ShallowWrapper | ReactWrapper, event = {}): void {
  element.simulate('click', { ...mockEvent, ...event });
}

export function clickOutside(event = {}): void {
  const dispatchedEvent = new MouseEvent('click', event);
  window.dispatchEvent(dispatchedEvent);
}

export function submit(element: ShallowWrapper): void {
  element.simulate('submit', {
    preventDefault() {}
  });
}

export function change(element: ShallowWrapper, value: string, event = {}): void {
  element.simulate('change', {
    target: { value },
    currentTarget: { value },
    ...event
  });
}

export function keydown(keyCode: number): void {
  const event = new KeyboardEvent('keydown', { keyCode } as KeyboardEventInit);
  document.dispatchEvent(event);
}

export function elementKeydown(element: ShallowWrapper, keyCode: number): void {
  const event = {
    currentTarget: { element },
    keyCode,
    preventDefault() {}
  };

  if (typeof element.type() === 'string') {
    // `type()` is string for native dom elements
    element.simulate('keydown', event);
  } else {
    element.prop<Function>('onKeyDown')(event);
  }
}

export function doAsync(fn?: Function): Promise<void> {
  return new Promise(resolve => {
    setImmediate(() => {
      if (fn) {
        fn();
      }
      resolve();
    });
  });
}

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider({ locale: 'en' }, {});
const { intl } = intlProvider.getChildContext();
export function shallowWithIntl(node: React.ReactElement<any>, options: ShallowRendererProps = {}) {
  return shallow(node, { ...options, context: { intl, ...options.context } });
}
