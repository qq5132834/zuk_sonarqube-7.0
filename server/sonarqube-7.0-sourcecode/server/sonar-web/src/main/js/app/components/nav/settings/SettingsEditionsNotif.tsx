/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import NavBarNotif from '../../../../components/nav/NavBarNotif';
import RestartForm from '../../../../components/common/RestartForm';
import { dismissErrorMessage, Edition, EditionStatus } from '../../../../api/marketplace';
import { translate, translateWithParameters } from '../../../../helpers/l10n';

interface Props {
  editions?: Edition[];
  editionStatus: EditionStatus;
  preventRestart: boolean;
  setEditionStatus: (editionStatus: EditionStatus) => void;
}

interface State {
  openRestart: boolean;
}

export default class SettingsEditionsNotif extends React.PureComponent<Props, State> {
  state: State = { openRestart: false };

  handleOpenRestart = () => this.setState({ openRestart: true });
  hanleCloseRestart = () => this.setState({ openRestart: false });

  handleDismissError = () =>
    dismissErrorMessage().then(
      () => this.props.setEditionStatus({ ...this.props.editionStatus, installError: undefined }),
      () => {}
    );

  renderStatusMsg(edition?: Edition) {
    const { editionStatus } = this.props;
    return (
      <NavBarNotif className="alert alert-info">
        <i className="spinner spacer-right text-bottom" />
        <span>
          {edition
            ? translateWithParameters(
                'marketplace.status_x.' + editionStatus.installationStatus,
                edition.name
              )
            : translate('marketplace.status', editionStatus.installationStatus)}
        </span>
      </NavBarNotif>
    );
  }

  renderRestartMsg(edition?: Edition) {
    const { editionStatus, preventRestart } = this.props;
    return (
      <NavBarNotif className="alert alert-success">
        <span>
          {edition
            ? translateWithParameters(
                'marketplace.status_x.' + editionStatus.installationStatus,
                edition.name
              )
            : translate('marketplace.status', editionStatus.installationStatus)}
        </span>
        {edition &&
          edition.key === 'datacenter' && (
            <span className="little-spacer-left">
              <FormattedMessage
                defaultMessage={translate('marketplace.see_documentation_to_enable_cluster')}
                id="marketplace.see_documentation_to_enable_cluster"
                values={{
                  url: (
                    <a
                      href="https://redirect.sonarsource.com/doc/data-center-edition.html"
                      target="_blank">
                      {edition.name}
                    </a>
                  )
                }}
              />
            </span>
          )}
        {!preventRestart && (
          <button className="js-restart spacer-left" onClick={this.handleOpenRestart}>
            {translate('marketplace.restart')}
          </button>
        )}
        {!preventRestart &&
          this.state.openRestart && <RestartForm onClose={this.hanleCloseRestart} />}
      </NavBarNotif>
    );
  }

  renderManualMsg(edition?: Edition) {
    const { editionStatus } = this.props;
    return (
      <NavBarNotif className="alert alert-danger">
        {edition
          ? translateWithParameters(
              'marketplace.status_x.' + editionStatus.installationStatus,
              edition.name
            )
          : translate('marketplace.status', editionStatus.installationStatus)}
        <a
          className="spacer-left"
          href={
            edition && edition.key === 'datacenter'
              ? 'https://redirect.sonarsource.com/doc/data-center-edition.html'
              : 'https://redirect.sonarsource.com/doc/how-to-install-an-edition.html'
          }
          target="_blank">
          {translate('marketplace.how_to_install')}
        </a>
        {edition && (
          <a
            className="button spacer-left"
            download={`sonarqube-${edition.name}.zip`}
            href={edition.downloadUrl}
            target="_blank">
            {translate('marketplace.download_package')}
          </a>
        )}
      </NavBarNotif>
    );
  }

  renderStatusAlert() {
    const { currentEditionKey, installationStatus, nextEditionKey } = this.props.editionStatus;
    const nextEdition =
      this.props.editions && this.props.editions.find(edition => edition.key === nextEditionKey);
    const currentEdition =
      this.props.editions &&
      this.props.editions.find(
        edition =>
          edition.key === currentEditionKey || (!currentEditionKey && edition.key === 'community')
      );

    switch (installationStatus) {
      case 'AUTOMATIC_IN_PROGRESS':
        return this.renderStatusMsg(nextEdition);
      case 'AUTOMATIC_READY':
        return this.renderRestartMsg(nextEdition);
      case 'UNINSTALL_IN_PROGRESS':
        return this.renderRestartMsg(currentEdition);
      case 'MANUAL_IN_PROGRESS':
        return this.renderManualMsg(nextEdition);
    }
    return null;
  }

  render() {
    const { installError } = this.props.editionStatus;
    if (installError) {
      return (
        <NavBarNotif className="alert alert-danger" onCancel={this.handleDismissError}>
          {installError}
        </NavBarNotif>
      );
    }

    return this.renderStatusAlert();
  }
}
