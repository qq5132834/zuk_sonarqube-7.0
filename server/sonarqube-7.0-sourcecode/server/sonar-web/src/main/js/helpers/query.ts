/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { isNil, omitBy } from 'lodash';
import { isValidDate, parseDate, toNotSoISOString } from './dates';

export interface RawQuery {
  [x: string]: any;
}

function arraysEqual(a: RawQuery, b: RawQuery): boolean {
  if (a.length !== b.length) {
    return false;
  }
  for (let i = 0; i < a.length; i++) {
    if (a[i] !== b[i]) {
      return false;
    }
  }
  return true;
}

export function queriesEqual(a: RawQuery, b: RawQuery): boolean {
  const keysA = Object.keys(a);
  const keysB = Object.keys(b);

  if (keysA.length !== keysB.length) {
    return false;
  }

  return keysA.every(
    key =>
      Array.isArray(a[key]) && Array.isArray(b[key])
        ? arraysEqual(a[key], b[key])
        : a[key] === b[key]
  );
}

export function cleanQuery(query: RawQuery): RawQuery {
  return omitBy(query, isNil);
}

export function parseAsBoolean(value: string | undefined, defaultValue: boolean = true): boolean {
  return value === 'false' ? false : value === 'true' ? true : defaultValue;
}

export function parseAsDate(value?: string): Date | undefined {
  if (value) {
    const date = parseDate(value);
    if (isValidDate(date)) {
      return date;
    }
  }
  return undefined;
}

export function parseAsFacetMode(facetMode: string): string {
  return facetMode === 'debt' || facetMode === 'effort' ? 'effort' : 'count';
}

export function parseAsString(value: string | undefined): string {
  return value || '';
}

export function parseAsArray(
  value: string | undefined,
  itemParser: (x: string) => string
): string[] {
  return value ? value.split(',').map(itemParser) : [];
}

export function serializeDate(value?: Date): string | undefined {
  if (value != null && value.toISOString) {
    return toNotSoISOString(value);
  }
  return undefined;
}

export function serializeString(value: string): string | undefined {
  return value || undefined;
}

export function serializeStringArray(value: string[] | undefined[]): string | undefined {
  return value && value.length ? value.join() : undefined;
}
