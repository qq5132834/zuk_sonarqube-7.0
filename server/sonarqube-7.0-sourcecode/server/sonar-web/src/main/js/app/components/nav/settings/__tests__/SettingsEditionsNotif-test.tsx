/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
/* eslint-disable import/order */
import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { click } from '../../../../../helpers/testUtils';
import SettingsEditionsNotif from '../SettingsEditionsNotif';

jest.mock('../../../../../api/marketplace', () => ({
  dismissErrorMessage: jest.fn(() => Promise.resolve())
}));

const dismissMsg = require('../../../../../api/marketplace').dismissErrorMessage as jest.Mock<any>;

beforeEach(() => {
  dismissMsg.mockClear();
});

it('should display an in progress notif', () => {
  const wrapper = shallow(
    <SettingsEditionsNotif
      editionStatus={{ installationStatus: 'AUTOMATIC_IN_PROGRESS' }}
      preventRestart={false}
      setEditionStatus={jest.fn()}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should display a ready notification', () => {
  const wrapper = shallow(
    <SettingsEditionsNotif
      editionStatus={{ installationStatus: 'AUTOMATIC_READY' }}
      preventRestart={false}
      setEditionStatus={jest.fn()}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should display a manual installation notification', () => {
  const wrapper = shallow(
    <SettingsEditionsNotif
      editionStatus={{ installationStatus: 'MANUAL_IN_PROGRESS', nextEditionKey: 'foo' }}
      editions={[
        {
          key: 'foo',
          name: 'Foo',
          textDescription: 'Foo desc',
          downloadUrl: 'download_url',
          homeUrl: 'more_url',
          licenseRequestUrl: 'license_url'
        }
      ]}
      preventRestart={false}
      setEditionStatus={jest.fn()}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should display install errors', () => {
  const wrapper = shallow(
    <SettingsEditionsNotif
      editionStatus={{ installationStatus: 'AUTOMATIC_IN_PROGRESS', installError: 'Foo error' }}
      preventRestart={false}
      setEditionStatus={jest.fn()}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should allow to dismiss install errors', async () => {
  const setEditionStatus = jest.fn();
  const wrapper = mount(
    <SettingsEditionsNotif
      editionStatus={{ installationStatus: 'NONE', installError: 'Foo error' }}
      preventRestart={false}
      setEditionStatus={setEditionStatus}
    />
  );
  click(wrapper.find('button'));
  expect(dismissMsg).toHaveBeenCalled();
  await new Promise(setImmediate);
  expect(setEditionStatus).toHaveBeenCalledWith({
    installationStatus: 'NONE',
    installError: undefined
  });
});

it('should not display the restart button', () => {
  const wrapper = shallow(
    <SettingsEditionsNotif
      editionStatus={{ installationStatus: 'AUTOMATIC_READY' }}
      preventRestart={true}
      setEditionStatus={jest.fn()}
    />
  );
  expect(wrapper.find('button.js-restart').exists()).toBeFalsy();
});

it('should have a link to cluster documentation for datacenter edition', () => {
  const editions = [{ key: 'datacenter' }] as any;
  const wrapper = shallow(
    <SettingsEditionsNotif
      editions={editions}
      editionStatus={{ installationStatus: 'AUTOMATIC_READY', nextEditionKey: 'datacenter' }}
      preventRestart={false}
      setEditionStatus={jest.fn()}
    />
  );
  expect(wrapper.find('FormattedMessage').exists()).toBeTruthy();
});
