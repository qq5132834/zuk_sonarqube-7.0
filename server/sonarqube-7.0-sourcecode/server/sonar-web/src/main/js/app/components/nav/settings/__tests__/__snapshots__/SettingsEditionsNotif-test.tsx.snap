// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a manual installation notification 1`] = `
<NavBarNotif
  className="alert alert-danger"
>
  marketplace.status_x.MANUAL_IN_PROGRESS.Foo
  <a
    className="spacer-left"
    href="https://redirect.sonarsource.com/doc/how-to-install-an-edition.html"
    target="_blank"
  >
    marketplace.how_to_install
  </a>
  <a
    className="button spacer-left"
    download="sonarqube-Foo.zip"
    href="download_url"
    target="_blank"
  >
    marketplace.download_package
  </a>
</NavBarNotif>
`;

exports[`should display a ready notification 1`] = `
<NavBarNotif
  className="alert alert-success"
>
  <span>
    marketplace.status.AUTOMATIC_READY
  </span>
  <button
    className="js-restart spacer-left"
    onClick={[Function]}
  >
    marketplace.restart
  </button>
</NavBarNotif>
`;

exports[`should display an in progress notif 1`] = `
<NavBarNotif
  className="alert alert-info"
>
  <i
    className="spinner spacer-right text-bottom"
  />
  <span>
    marketplace.status.AUTOMATIC_IN_PROGRESS
  </span>
</NavBarNotif>
`;

exports[`should display install errors 1`] = `
<NavBarNotif
  className="alert alert-danger"
  onCancel={[Function]}
>
  Foo error
</NavBarNotif>
`;
